R/GeospatialFunctions.R

```diff
@@ -0,0 +1,612 @@
+    
+    # LINE FEATURES - try to pull line AU data if it exists. Otherwise, move on...
+    ATTAINS_lines <- NULL
+    try(ATTAINS_lines <- fetchATTAINS(type = "lines", data = TADA_DataRetrieval_data) %>%
+          # subset to only ATTAINS line features in the same NHD HR catchments as WQP observations
+          .[nearby_catchments,] %>%
+          # make sure no duplicate line features exist
+          dplyr::distinct(assessmentunitidentifier, .keep_all = TRUE),
+        silent = TRUE)
+    
+    # POLYGON FEATURES - try to pull polygon AU data if it exists. Otherwise, move on...
+    ATTAINS_polygons <- NULL
+    try(ATTAINS_polygons <- fetchATTAINS(type = "polygons", data = TADA_DataRetrieval_data) %>%
+          # subset to only ATTAINS polygon features in the same NHD HR catchments as WQP observations
+          .[nearby_catchments,] %>%
+          # make sure no duplicate polygon features exist
+          dplyr::distinct(assessmentunitidentifier, .keep_all = TRUE),
+        silent = TRUE)
+  }))
+  
+  return(list("TADA_with_ATTAINS" = TADA_with_ATTAINS,
+              "ATTAINS_catchments" = ATTAINS_catchments,
+              "ATTAINS_points" = ATTAINS_points,
+              "ATTAINS_lines" = ATTAINS_lines,
+              "ATTAINS_polygons" = ATTAINS_polygons))
+}
+
+}
+
+
+#' TADA_ViewATTAINS
+#' 
+#' Visualizes the data returned from TADA_GetATTAINS if return was set to TRUE. 
+#'
+#' @param ATTAINS_list A list containing a data frame and ATTAINS shapefile objects created by `TADA_GetATTAINS()` with the return argument set to TRUE. 
+#' 
+#' @return A leaflet map visualizing the TADA water quality observations and the linked ATTAINS assessment units. All maps are in WGS84.
+#' 
+#' @seealso [TADA_DataRetrieval()]
+#' @seealso [TADA_GetATTAINS()]
+#' 
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#'tada_data <- TADA_DataRetrieval(startDate = "1990-01-01",
+#'                                endDate = "1995-12-31",
+#'                                characteristicName = "pH",
+#'                                statecode = "NV",
+#'                                applyautoclean = TRUE)
+#'                                
+#'attains_data <- TADA_GetATTAINS(data = tada_data, return = TRUE)                               
+#'                                  
+#'TADA_ViewATTAINS(ATTAINS_list = attains_data)
+#' }
+
+TADA_ViewATTAINS <- function(ATTAINS_list){
+  
+  if(!all(c("TADA_with_ATTAINS", "ATTAINS_catchments", "ATTAINS_points",
+            "ATTAINS_lines", "ATTAINS_polygons") %in% names(ATTAINS_list))) {
+    stop("Your ATTAINS_list was not produced from `TADA_GetATTAINS()`. Please
+         create your list of ATTAINS features using `TADA_GetATTAINS()`.")
+  }
+  
+  data <- ATTAINS_list[["TADA_with_ATTAINS"]]
+  ATTAINS_catchments <- ATTAINS_list[["ATTAINS_catchments"]]
+  ATTAINS_points <- ATTAINS_list[["ATTAINS_points"]]
+  ATTAINS_lines <- ATTAINS_list[["ATTAINS_lines"]]
+  ATTAINS_polygons <- ATTAINS_list[["ATTAINS_polygons"]]
```

Comment:
thought: This reminded me of the existence of `list2env()`, which could be an alternative for this in the future. However given that I'm not sure if/how much it could slow the function down and the fact I haven't used it more than once or twice I'm not sure I'd actually recommend it yet.